### PR TITLE
Add support for basic auth in helm

### PR DESCRIPTION
**Pull request by machacekondra**
Fixes: https://github.com/stolostron/cluster-templates-operator/issues/65

Signed-off-by: Ondra Machacek <omachace@redhat.com>

### CODE REVIEW

**Comment by rawagner** [comment]
@machacekondra It would be great if `GetRepoHTTPClient` already returns an http client which, for every request, adds credentials. Do you think it would be a good approach to modify the `Do` function of `http.Client` so it adds the credentials ? That way we wont have to call `CreateRequest` separately before the `Do`.

**Comment by machacekondra** [comment]
> @machacekondra It would be great if `GetRepoHTTPClient` already returns an http client which, for every request, adds credentials. Do you think it would be a good approach to modify the `Do` function of `http.Client` so it adds the credentials ? That way we wont have to call `CreateRequest` separately before the `Do`.

Done

File: helm/client.go [hunk 1]
```diff
@@ -206,5 +239,36 @@ func GetRepoHTTPClient(
...
-	return httpClient, nil
+	client := &HttpClient{
+		secret: repoSecret,
+		client: httpClient,
+	}
+
+	return client, nil
+}
+
+func CreateRequest(
```
Comment (rawagner) [inline helm/client.go:250]: we dont need this function anymore

**Review by rawagner — CHANGES_REQUESTED** [review]

**Pull request merged**